PL-32506 NG IP Allowlist (#1581)
harness/developer-hub: docs/platform/20_References/allowlist-harness-domains-and-ips.md

:::

Management of the IP allowlist involves reviewing and updating the list of approved IP addresses regularly to ensure that only authorized users are granted access. After you add and enable an IP allowlist, only users connecting from the IP address/CIDR (Classless Inter-Domain Routing) block may log into Harness.

Following are some key benefits of IP allowlist management:

- Increased security: By limiting access to approved IP addresses only, you can prevent unauthorized access to sensitive data and resources. This protects against data breaches, cyber-attacks, and other security threats.

- Improved network performance: By limiting access to approved IP addresses only, you can reduce the number of requests your network receives, which improves network speed and reduces latency.

:::important
Harness IP allowlisting is for user connections to a Harness account. It will not perform IP allowlisting for Harness Delegate connections.
:::

This topic explains how to set up IP allowlist in Harness.

## Important
- Make sure you have **View**, **Create/Edit**, and **Delete** permissions for **Authentication Settings** to add and manage IP allowlist.

> - Harness throws HTTP 401 errors if an IP address is not allowlisted but tries to access Harness.

Reviewer: ```suggestion
- Harness throws HTTP 401 error if an IP address is not allowlisted but tries to access Harness.
```

- Increased security: By limiting access to approved IP addresses only, you can prevent unauthorized access to sensitive data and resources. This protects against data breaches, cyber-attacks, and other security threats.

- Improved network performance: By limiting access to approved IP addresses only, you can reduce the number of requests your network receives, which improves network speed and reduces latency.

:::important
Harness IP allowlisting is for user connections to a Harness account. It will not perform IP allowlisting for Harness Delegate connections.
:::

This topic explains how to set up IP allowlist in Harness.

## Important
- Make sure you have **View**, **Create/Edit**, and **Delete** permissions for **Authentication Settings** to add and manage IP allowlist.
- Harness returns a status code of HTTP 401 if an IP address is not allowlisted but tries to access Harness.
- An allowlist can be based on IPv4 or IPv6 or a mix of both.
- Following are the exceptions to the IP allowlist in Harness:
  - Calls corresponding to SCIM.
  - Calls corresponding to Identity providers.
  - API calls corresponding to Harness admin.

> - Harness uses an in-memory cache. All caches expire after 15 minutes. Every update takes 15 minutes because there are no manual cache updates.

Reviewer: ```suggestion
- Harness uses an in-memory cache. All caches expire after 5 minutes. Every update takes 5 minutes because there are no manual cache updates.
```

> - Harness uses an in-memory cache. All caches expire after 15 minutes. Every update takes 15 minutes because there are no manual cache updates.

Reviewer: Cache timings are 5 minutes instead of 15.

## Add IP addresses in the allowlist

To add IP allowlist in Harness: 

1. Select **ACCOUNT SETTINGS** and then select **Authentication**.
2. Select **Allowlist**.
   
   ![](./static/allowlist-tab.png)

3. To add IP addresses, select **Add IP Addresses**.
   The **IP Allowlist** settings appear.
4. Enter **Name** for the allowlist.
5. Select **Continue**.

   ![](./static/allowlist-overview.png)

## Define the IP range

1. In **IP Address/CIDR**, enter a CIDR or an individual IP address or a block of IP addresses. 
2. You can select one of the following for the allowlist application: 
   - **UI**: Harness allows or blocks the IP address based on the UI request.
   - **API**: Harness checks the API key in the request and allows or blocks the IP address.
3. Select **Save and Continue**.

   ![](./static/allowlist-define-range.png)
  
## Test IP

> 1. Enter an IP address or a block of IP addresses to check if they are allowed.

Reviewer 2: Enter an IP address to check if it falls in IP address range configured in previous step.

2. Select **Test**.
3. Select **Finish**.

   ![](./static/allowlist-test-ip.png)

   You can create multiple allowlists and enable or disable them as needed.

## Manage IP allowlist

The IP addresses configured in the allowlist now appear in the **IP Allowlist**.

> 1. Toggle enable or disable beside the allowlist to enable or disable an allowlist.

Reviewer 2: Should we add a note/warning to this for customers?
`Make sure your current IP is allowlisted in all enabled configs, else they can get locked out` ?

helpdocs_is_published: true
---

> Harness SaaS Delegates only need outbound access to the Harness domain name (most commonly, **app.harness.io**) and, optionally, to **logging.googleapis.com** (used to provide logs to Harness support).

Reviewer 3: ```suggestion
Harness SaaS Delegates only need outbound access to the Harness domain name (typically **app.harness.io**) and, optionally, to **logging.googleapis.com** (used to provide logs to Harness support).
```

Reviewer 3: trying to lessen the amount of commas :-)

The URL `logging.googleapis.com` is used to send log files to Harness support. Use the environment variable `STACK_DRIVER_LOGGING_ENABLED` to disable this functionality.

For more information, go to [Delegate environment variables](/docs/platform/delegates/delegate-reference/delegate-environment-variables/#stack_driver_logging_enabled).

## Harness Manager

Users of the Harness Manager browser client need access to **app.harness.io** and **static.harness.io**. This is not a Harness Delegate requirement. It's simply for users to use the browser-based Harness Manager.

## Vanity URL

If you are using a Harness vanity URL, like **mycompany.harness.io**, you can allowlist it also.

## Allowlist Harness SaaS IPs

The following list is optional. You can allowlist these IPs if needed.

```
35.201.91.229  
162.159.134.64  
162.159.135.64  
2606:4700:7::a29f:8640  
2606:4700:7::a29f:8740
```

:::caution

> Harness will not change IPs without 30 days notice to all customers. If a security emergency requires a change, all customers will be notified.

Reviewer 3: ```suggestion
Harness will not change IPs without 30 days notice to all customers. If a security emergency requires a change, all customers are notified.
```

helpdocs_is_published: true
---

:::important
Currently, this feature is behind the feature flag `PL_IP_ALLOWLIST_NG`. Contact Harness Support to enable the feature.
:::

Management of the IP allowlist involves reviewing and updating the list of approved IP addresses regularly to ensure that only authorized users are granted access. After you add and enable an IP allowlist, only users connecting from the IP address/CIDR (Classless Inter-Domain Routing) block may log into Harness.

> Following are some key benefits of IP whitelist management:

Reviewer: ```suggestion
Following are some key benefits of IP allowlist management:
```
